#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material2 "slippy"
      coulombFriction [
        0
      ]
    }
    ContactProperties {
      material2 "block"
      coulombFriction [
        0.01
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5456484755477622 0.7681687330521343 0.33493960454607624 1.126352793899625
  position 3.2634359294878768 2.614878781053615 1.5558621459750852
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 2.4 2.4
  appearance Asphalt {
  }
}
DEF Block_R1 Solid {
  translation 0.377 0.05 -0.9
  children [
    DEF BOX_R Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "solid(4)"
  contactMaterial "box"
  boundingObject USE BOX_R
  physics Physics {
  }
}
DEF Block_R2 Solid {
  translation 1 0.05 0.3
  children [
    USE BOX_R
  ]
  name "solid(13)"
  contactMaterial "block"
  boundingObject USE BOX_R
  physics Physics {
  }
}
DEF Block_R3 Solid {
  translation -1 0.05 -0.5
  children [
    USE BOX_R
  ]
  name "solid(12)"
  boundingObject USE BOX_R
  physics Physics {
  }
}
DEF Block_R4 Solid {
  translation 0.1 0.05 1.1
  children [
    USE BOX_R
  ]
  name "solid(7)"
  boundingObject USE BOX_R
  physics Physics {
  }
}
DEF Block_B1 Solid {
  translation -1.1 0.05 1
  children [
    DEF BOX_B Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "solid(8)"
  boundingObject USE BOX_B
  physics Physics {
  }
}
DEF Block_B2 Solid {
  translation 0.03 0.05 0.72
  children [
    USE BOX_B
  ]
  name "solid(9)"
  boundingObject USE BOX_B
  physics Physics {
  }
}
DEF Block_B3 Solid {
  translation -0.85 0.05 -0.2
  children [
    USE BOX_B
  ]
  name "solid(10)"
  boundingObject USE BOX_B
  physics Physics {
  }
}
DEF Block_B4 Solid {
  translation 1 0.05 0.7
  children [
    USE BOX_B
  ]
  name "solid(11)"
  boundingObject USE BOX_B
  physics Physics {
  }
}
DEF Floor_colour Solid {
  translation 1 0.0001 1
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(6)"
}
DEF Floor_colour Solid {
  translation 1 0.0001 -1
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(5)"
}
DEF WALL_1 Solid {
  translation 1.2 0.05 0
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 2.38
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -1.2 0.05 0
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
DEF NEW_ROBOT Robot {
  translation -0.42 0.09 0.42
  children [
    Compass {
    }
    Camera {
      translation 0 0 0.1
      width 1
      height 1
    }
    DEF CASTER Solid {
      translation 0 -0.065 -0.1
      children [
        Shape {
          geometry DEF CASTOR_BALL Sphere {
            radius 0.01
          }
        }
      ]
      name "solid(2)"
      contactMaterial "slippy"
      boundingObject USE CASTOR_BALL
      physics Physics {
      }
    }
    DEF BODY Group {
      children [
        DEF TOP Shape {
          appearance DEF BLUE PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.01 0.2
          }
        }
        DEF LEFT_SIDE Transform {
          translation 0.045 -0.035 0
          children [
            DEF SIDE Shape {
              appearance USE BLUE
              geometry Box {
                size 0.01 0.06 0.2
              }
            }
          ]
        }
        DEF RIGHT_SIDE Transform {
          translation -0.045 -0.035 0
          children [
            USE SIDE
          ]
        }
        DEF BACK Transform {
          translation 0 -0.035 -0.095
          children [
            DEF BACK Shape {
              appearance USE BLUE
              geometry Box {
                size 0.08 0.06 0.01
              }
            }
          ]
        }
      ]
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 -0.04 0.06
      }
      device [
        RotationalMotor {
          name "wheel1"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.06 -0.04 0.06
        rotation 0 0 1 1.5708036407833588
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 -0.04 0.06
      }
      device [
        RotationalMotor {
          name "wheel2"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.06 -0.04 0.06
        rotation 0 0 1 1.5708029067519502
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
    GPS {
      translation 0 0.005 0
    }
  ]
  name "blueRobot"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "go_home_controller"
}
Robot {
  translation 1 0.05 1
  rotation 0 1 0 3.14
  children [
    Camera {
      translation 0 0 0.1
      width 1
      height 1
    }
    Solid {
      translation 0 -0.03 0
      children [
        Shape {
          geometry DEF CASTOR_BALL Sphere {
            radius 0.01
          }
        }
      ]
      name "solid(2)"
      contactMaterial "slippy"
      boundingObject USE CASTOR_BALL
      physics Physics {
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.06000000000005251 1.4531748733392746e-07 0.04999909020156158
        rotation 0.0019079727809748464 -0.0019079797047584805 -0.9999963596200304 4.712381666377444
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.05999999999986491 2.9408924358078494e-07 0.049999115915629125
        rotation 0.0017048106017812177 -0.001704816680961621 -0.9999970936062246 4.712382400408873
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  name "redRobot"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "red_controller"
}
NewRRobot {
}
